### ISSUE TITLE
Custom Rule is not loaded - no errors

### ISSUE THREAD

**Issue by moseleyi** [sqlfluff/sqlfluff#6041]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I developed a simple custom rule and it's not being loaded.

### Expected Behaviour

The sample code fails the rule, therefore the output of the whole linting command would be FAIL

### Observed Behaviour

The script passes the linting

### How to reproduce

My rule definition `custom_rules.py` is at the same level as `.sqlfluff` file:
```python
from sqlfluff.core.rules import BaseRule, LintResult
from sqlfluff.core.parser import BaseSegment

class Rule_LL99(BaseRule):
    """Ensure columns with functions are aliased"""

    def _eval(self, segment: BaseSegment, **kwargs):
        print(f"Evaluating segment: {segment}")

        if segment.is_type('function'):
            parent_segment = segment.parent
            if parent_segment and not any(child.is_type('alias_expression') for child in parent_segment.segments):
                return LintResult(anchor=segment, description="Columns where Function has been used must be aliased")
        return None
```

SQL being tested
```sql
SELECT
  id,
  status,
  LOWER("text"),
  CAST("test" AS STRING)
FROM test
```

Place where I load my rules
```
[sqlfluff:rules]
custom_rules = custom_rules::Rule_LL99
```

When running the test
```
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
All Finished 📜 🎉!
```

I tried to develop a python script to test the rule on its own and I got some issues like wrong name of the rule - something that I would prefer to see during linting, otherwise it's hard to know what's going on.

I'm not a python developer, hence I wonder if there's anything obvious.

### Dialect

bigquery

### Version

SQLFluff: 3.10
Python: 3.12.2

### Configuration

```
[sqlfluff]
templater = dbt
dialect = bigquery
exclude_rules = RF02, AL01, AL05, ST07, AL03

[sqlfluff:templater:dbt]
project_dir = ./

[sqlfluff:indentation]
tab_space_size = 2
indented_joins = true
indented_using_on = false

[sqlfluff:rules]
custom_rules = custom_rules::Rule_LL99

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:rules:aliasing.length]
min_alias_length = 3
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by pikachuev** [comment]
Any suggestions?

**Comment by ltchikindas** [comment]
We're experiencing the same issue. @moseleyi  were you able to resolve / work around?

**Comment by ltchikindas** [comment]
Following up here now that I've had a moment to deal with ours.

I think there are 2 problems:
1. SQLFluff is pedantic about how rules are named, down to the character counts after the `_`
2. You have to show it where the rules are by running a pip install

I don't think 
```
[sqlfluff:rules]
custom_rules = custom_rules::Rule_LL99
```
does anything...

How I went about it: 

Copy in the files from  [here](https://github.com/sqlfluff/sqlfluff/tree/main/plugins/sqlfluff-plugin-example
) into your own project directory

Meticulously rename all of the mentions throughout. There are 5  strings to look out for, spread between file names or inside of files:

"Rule_Example_L001"
"Example_L001"
"sqlfluff_plugin_example"
"sqlfluff-plugin-example"
"sqlfluff_example"

I used
"Rule_IndentFirstLine_CU01"
"IndentFirstLine_CU01
"sqlfluff_plugin_indent_first_line"
"sqlfluff-plugin-indent-first-line"
"sqlfluff_indent_first_line"

and then ran 

```
pip install -e sqlfluff/plugins/sqlfluff-plugin-indent-first-line
```


Good luck!

**Comment by peterbud** [comment]
Hi @moseleyi  thanks for opening this issue.

@ltchikindas is right - you need the source install the plugin with 'pip install -e'. More you can found here: https://docs.sqlfluff.com/en/stable/guides/setup/developing_custom_rules.html
